cgrep.py

```diff
@@ -0,0 +1,443 @@
+
+
+def cli_options():
+  """ builds the argparse options for cgrep.
+
+  :returns argparse.ArgumentParser: the arguments, ready to be parsed.
+  """
+
+  parser = argparse.ArgumentParser(
+      description="c[apirca]grep",
+      formatter_class=argparse.RawTextHelpFormatter
+  )
+
+  parser.add_argument("-d", "--def", dest="defs",
+                      help="Network Definitions directory location. \n",
+                      default="./def")
+
+  # -i and -t can be used together, but not with any other option.
+  ip_group = parser.add_argument_group()
+  # take 1 or more IPs
+  ip_group.add_argument("-i", "--ip", dest="ip", nargs='+', type=is_valid_ip,
+                        help="Return list of definitions containing the "
+                        "IP(s).\nMultiple IPs permitted.")
+
+  ip_group.add_argument("-t", "--token", dest="token",
+                        help=("See if an IP is contained within the given "
+                              "token.\nMust be used in conjunction with "
+                              "-i/--ip [addr]."))
+
+  exclusive_group = parser.add_mutually_exclusive_group()
+  # the rest of the arguments are mutually exclusive with each other,
+  # and -i / -t
+  exclusive_group.add_argument("-c", "--cmp", dest="cmp", nargs=2,
+                               metavar=("OBJ", "OBJ"),
+                               help=("Compare the two given network "
+                                     "definition tokens"))
+
+  exclusive_group.add_argument("-g", "--gmp", dest="gmp", nargs=2,
+                               type=is_valid_ip, metavar=("IP", "IP"),
+                               help=("Diff the network objects to"
+                                     " which the given IP(s) belong"))
+
+  exclusive_group.add_argument("-o", "--obj", dest="obj", nargs='+',
+                               help=("Return list of IP(s) contained within "
+                                     "the given token(s)"))
+
+  exclusive_group.add_argument("-s", "--svc", dest="svc", nargs='+',
+                               help=("Return list of port(s) contained "
+                                     "within given token(s)"))
+
+  exclusive_group.add_argument("-p", "--port", dest="port", nargs=2,
+                               metavar=('PORT', 'PROTO'),
+                               help=("Returns a list of tokens containing "
+                                     "the given port and protocol"))
+
+  return parser
+
+
+def main(parser):
+  """ Determines the code path based on the arguments passed.
+
+  :parser argparse.ArgumentParser: the argument parser, but not parsed yet.
+  :returns None:
+  """
+  options = parser.parse_args()
+  db = naming.Naming(options.defs)
+  p = pprint.PrettyPrinter(indent=1, depth=4, width=1).pprint
+
+  # if -i and any other option:
+  if options.ip and any([options.gmp, options.cmp, options.obj, options.svc,
+                         options.port]):
+    print("You can only use -i with -t or by itself")
```

Comment:
Have you thought about using logging instead of simple print statements? Also the parens are superfluous, I won't force you to change them though. Give logging some thought though.